src/cls/rbd/cls_rbd.cc

```diff
@@ -4428,6 +4514,89 @@ int mirror_instances_remove(cls_method_context_t hctx, bufferlist *in,
 }
 
 /**
+ * Input:
+ * @param start_after: key to start after
+ * @param max_return: max return items
+ *
+ * Output:
+ * @param std::map<std::string, cls::rbd::MirrorImageMap>: image mapping
+ * @returns 0 on success, negative error code on failure
+ */
+int mirror_image_map_list(cls_method_context_t hctx, bufferlist *in,
+                          bufferlist *out) {
+  std::string start_after;
+  uint64_t max_return;
+  try {
+    bufferlist::iterator it = in->begin();
+    ::decode(start_after, it);
+    ::decode(max_return, it);
+  } catch (const buffer::error &err) {
+    return -EINVAL;
+  }
+
+  std::map<std::string, cls::rbd::MirrorImageMap> image_mapping;
+  int r = mirror::mirror_image_map_list(hctx, start_after, max_return, &image_mapping);
+  if (r < 0) {
+    return r;
+  }
+
+  ::encode(image_mapping, *out);
+  return 0;
+}
+
+/**
+ * Input:
+ * @param std::map<std::string, cls::rbd::MirrorImageMap>: image mapping
```

Comment:
I'm actually ok with both. If the overhead of back-to-back exec (1024 max at a time) calls is not much, then lets choose that just to keep things in sync w/ rest of the cls rbd codebase?

I can update this by tomorrow if we finalize this interface.